src/common/transformations/include/transformations/common_optimizations/align_mixed_fp32_fp16_types.hpp

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/pass/graph_rewrite.hpp"
+#include "openvino/pass/pass.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+namespace pass {
+
+class TRANSFORMATIONS_API AlignMixedFP32FP16Types;
+
+}  // namespace pass
+}  // namespace ov
+
+/**
+ * @ingroup ie_transformation_common_api
+ * @brief AlignMixedFP32FP16Types adds Converts to keep mixed FP16/FP32 graph type consistent
```

Comment:
Why do we need such transformation? In what cases the graph can become type inconsistent?

Reply:
In many cases. E.g. we have elementwise op, let's say an `Add` one input of which was marked as to be kept in fp32 while other is not. In that case we need to insert convert from f16 into f32 to the second input.

Every time we face a node that has input without marking and node itself is marked with `disable_fp16_compression` this means we entered precision sensitive subgraph and we need to convert from `f16 -> f32` on it's `inputs`. And every time the node itself is marked as `disable_fp16_compression` but it's outputs are not we need to insert Convert from `f32 -> f16` on node `outputs`.

Reply:
Why didn't we pass other path for the second operand to mark it fp32 sensitive?